RELEASE.md

```diff
@@ -4,7 +4,11 @@ We try to follow the practice of releasing often. That allows us to have smaller
 
 ## Creating a release
 
-Creating a release happens via GitHub Actions by creating an [annotated](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_creating_tags) git tag. Tag creation triggers the release workflow which will do most of the heavy-lifting:
+Creating a release happens via GitHub Actions by creating an [annotated](https://git-scm.com/book/en/v2/Git-Basics-Tagging#_creating_tags) git tag.
+
+**WARNING**: The tag cannot be pushed with `git push --tags` because it won't trigger the [release GitHub Action](https://github.com/k0sproject/k0s/actions/workflows/release.yml). You must do `git push <tag>`.
+
+Tag creation triggers the release workflow which will do most of the heavy-lifting:
```

Comment:
I would also add the explicit example of annotated tag (I just today forgot to make the release tag annotated lol)